mongoengine/dereference.py

```diff
@@ -88,15 +88,15 @@ def _find_references(self, items, depth=0):
         if not items or depth >= self.max_depth:
             return reference_map
 
-        # Determine the iterator to use
-        if not hasattr(items, 'items'):
-            iterator = enumerate(items)
+        # Determine the iterable to use
+        if hasattr(items, 'values'):
+            iterable = items.itervalues()
```

Comment:
Well, 2to3 takes care of that: https://github.com/MongoEngine/mongoengine/blob/79a2d715b03cbb61ad4fbbac8eae2de27edb5466/setup.py#L54
You might verify that it gets translated to `iterable = iter(items.values)` as mentioned in https://www.python.org/dev/peps/pep-0469/